### DISCUSSION TITLE
'docker not found' after 'brew install --cask docker'

### DISCUSSION THREAD

**Discussion by nva14** [Homebrew/discussions#3471] [Casks]
I installed Docker Desktop for Mac (apple silicon) using Homebrew, but docker command not available after installation. Computer restarted.

.zshrc content:

if type brew &>/dev/null
then
  FPATH="$(brew --prefix)/share/zsh/site-functions:${FPATH}"

  autoload -Uz compinit
  compinit
fi

Which step did i miss?

Thanks for any help :)

**Comment by SMillerDev** [comment]
Well, casks do not install to `HOMEBREW_PREFIX`. They install wherever they want. In the case of Docker it installs to `/usr/local/bin`

**Reply by nva14** [reply]
That's odd, i can't find anything in folder `/usr/local/bin`. I added `export PATH=/usr/local/bin:$PATH` to .zshrc and that does nothing.

This computer previously had Docker Desktop for Mac installed by downloading from official Docker website, later i uninstalled it.

**Reply by osalbahr** [reply]
Yeah, it is strange. But the tools are only installed in `/usr/local/bin` *after* you open the app and go through the setup with default settings.

**Comment by galeselee** [comment]
[This issue](https://stackoverflow.com/questions/40523307/brew-install-docker-does-not-include-docker-engine) will be helpful

**Comment by spacether** [comment]
Docker CLI tools are installed under $HOME/.docker/bin -- so docker is not available on the CLI by default. You can update your PATH or go to Docker Desktop settings -> Advanced -> select "System (Requires password)"
This will automatically hoist the docker cli into your terminal environment.

By default docker only installs the cli binaries into $HOME/.docker/bin and the user must manually add that to their path.

**Reply by chitalian** [reply]
THIS!! Thank you for writing this

**Reply by chitalian** [reply]
`echo "export PATH=\"\$HOME/.docker/bin:\$PATH\"" >> ~/.zshrc`

Btw - A quick one-liner for anyone that wants to get up and running quickly without adding docker to your system

**Reply by osalbahr** [reply]
If I understand correctly, it only installs in `$HOME/.docker/bin` if you open the app and select User in the setup page.

**Comment by code-instable** [comment]
This is a solution which worked for me :

- after installing docker via `brew install --cask docker`, you need to finish the installation by running the Docker application ( `Docker.app`) which has been added to your app folder.

- Then you will be able to use it in the command line / terminal

the hint was : 

```
$ > brew reinstall --cask docker
==> Installing Cask docker
==> Moving App 'Docker.app' to '/Applications/Docker.app'
```

Hope it helps some

**Comment by lkapt** [comment]
The issue described is that the docker command suddenly stopped being recognized by the system. The following steps resolved the issue:

    Open the Docker Desktop application.
    Navigate to the "Advanced" settings.
    Locate the option labeled "Choose how to configure the installation of Docker's CLI tools."
    Temporarily change the configuration setting to "User."
    Afterward, revert the configuration setting back to "System."

These steps helped reconfigure the Docker CLI tools and restored the ability to use the docker command on the system.

**Comment by osalbahr** [comment]
I opened Homebrew/homebrew-cask#146078 specifically about this exact issue

For now, these workarounds seem to work:

First, run `brew uninstall --formula docker --force`. This is because the cask already comes with the CLI tools, just not in the right directory yet. So, you'd want to get rid of duplicates or possible conflicts. The command should also fix the error that occurs when trying to install the cask after having already installed the formula.

a) Open the app. Click on Troubleshoot (next to the Settings gear). Click Uninstall. Once you get the pop-up. Then, reopen the app. The setup page  should appear. You can either keep it as System or pick User, both should work
b) In Settings -> Advances, Switch to User then back to System in the Sr (as in https://github.com/orgs/Homebrew/discussions/3471#discussioncomment-5748926 as well)

**Comment by matikepa** [comment]
```
brew link docker
```
